File: src/main/java/practice/payment/application/in/PaymentService.java
package practice.payment.application.in;

import lombok.RequiredArgsConstructor;
import org.springframework.stereotype.Service;
import practice.payment.application.out.SavePaymentPort;
import practice.payment.domain.Payment;

@Service
@RequiredArgsConstructor
public class PaymentService implements ExternalPaymentUseCase {

  private final SavePaymentPort savePaymentPort;

  @Override
  public Payment prepare(Payment payment) {
    return savePaymentPort.saveToPrepare(payment);
  }

  @Override
  public Payment success(Payment payment) {
    return savePaymentPort.saveToSuccess(payment);
  }
}

File: src/main/kotlin/practice/payment/application/in/PaymentService.kt
package practice.payment.application.`in`

import org.springframework.stereotype.Service
import practice.payment.application.out.SavePaymentPort
import practice.payment.domain.Payment

@Service
class PaymentService(
    private val savePaymentPort: SavePaymentPort
) : ExternalPaymentUseCase {

    override fun prepare(payment: Payment): Payment {
        return savePaymentPort.saveToPrepare(payment)
    }

    override fun success(payment: Payment): Payment {
        return savePaymentPort.saveToSuccess(payment)
    }
}
